tests/neg/i11225.scala

```diff
@@ -0,0 +1,30 @@
+import compiletime.uninitialized
+
+class Test:
+
+  val x1: Int = uninitialized // error
+  var x2: Int = if ??? then uninitialized else uninitialized // error // error
+  var x3: Int = if true then uninitialized else 1 // error
+  var x4: Int = if false then uninitialized else 1 // ok, since inlined away
+  var x5: () => Int = () => uninitialized // error
+  var x6: Int = { uninitialized } // error
+
+  def f(x: Boolean) =
+    var cached: Int = uninitialized   // error
+    cached = if x then 1 else uninitialized  // error
+
+  var c: Int =
+    uninitialized // error
+    uninitialized // error
+    2
+
+  var d: Int =
+    println("pseudo init")
+    uninitialized // error
+
+  transparent inline def uni = uninitialized
+
+  inline def g(inline x: Int): Unit = ()
+  def f2 = g(uninitialized) // this one is ok since `uninitialized` is inlined away
+
+  var x7: Int = uni  // error
```

Comment:
It's an error since we end up with `Inlined(uninitialized)`. I think that's OK. Ideally we would not allow the inline def of `uni` in the first place.

Reply:
Then it would be inconsistent with `var x4: Int = if false then uninitialized else 1 // ok, since inlined away` and `  def f2 = g(uninitialized) // this one is ok since uninitialized is inlined away`. We would have some special inlining rules that only apply `uninitialized` which will confuse everyone.

Reply:
I think we need  not overthink this. The intention is that you should not use `uninitialized`  except as the RHS of a field definition. For all practical purposes that's what this PR achieves. We don't need to split hairs beyond that,